Add remill and gap portfiles

xlauko opened the pull request:
Add `remill` and `gap` ports. 

`remill` port allows to specify features `cxx-common-llvm-14` or `system-llvm` to build with local install `llvm` or uses `cxx-common` provided LLVM.

Commit:
ports: Add remill port.

Commit:
ports: Add gap library.

Commit:
ports: Add remill port features.

Commit:
ports: Fix remill features.

Commit:
ports: Fix redundent remill debug paths.

Commit:
ports: Update remill config paths.

Commit:
ports: Update gap port file configs.

Commit:
ports: Don't build gap tests.

Commit:
ports: Remove empty lib folder from gap port.

Commit:
ports: Add zlib to remill dependencies.

Commit:
ports: Fix config install paths.

Commit:
ports: Bump up remill port.

ekilmer committed:
Various fixes

Commit:
ports: bump up gap version

ekilmer committed:
Fix usage file

Previously, vcpkg would ignore it and just print the heuristically
generated usage.

ekilmer commented on ports/remill/portfile.cmake:
```diff
@@ -0,0 +1,37 @@
+vcpkg_from_github(
+  OUT_SOURCE_PATH SOURCE_PATH
+  REPO lifting-bits/remill
+  REF e7c0e3f9f7f482b6bcc336cd98b8afd4129c6e3b
+  SHA512 49ee8db4dbf097e89046f3fce3de85ea35703d329f44b3388af992664760044aa8a9e7d0dac1c71e914b0fe0293f5ec10cc8ef4de5e5a118a492b6b2581b9b8c
+  HEAD_REF vcpkg-manifest
```
I think I mentioned this already privately, but ideally we use a commit on `master` branch before merging this

Commit:
ports: bump up gap port

Commit:
ports: bump up remill

lkorenc committed:
ports: Bump remill.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/lifting-bits/cxx-common?pullRequest=950) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/lifting-bits/cxx-common?pullRequest=950) before we can accept your contribution.<br/>**3** out of **5** committers have signed the CLA.<br/><br/>:white_check_mark: ekilmer<br/>:white_check_mark: Ninja3047<br/>:white_check_mark: PappasBrent<br/>:x: Henrich Lauko<br/>:x: lkorenc<br/><hr/>**Henrich Lauko** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/lifting-bits/cxx-common?pullRequest=950) it.</sub>

Commit:
ports: bump up default llvm version to 16 for remill

Commit:
vcpkg: bump up remill portfile hash

Commit:
ports: bump up remill version decription

ekilmer committed:
Improve port upgrading experience (#1027)

Only upgrade ports that are specified. Also, ask the user if they're sure they want to upgrade.

It should prevent accidentally deleting/rebuilding LLVM if it's out of date or if trying to upgrade a port it depends on.

PappasBrent committed:
Combine unknown attrs annotate and attr type attrs (#1028)

- Replace patch `0027-unknown-attrs-as-annotations.patch` with a new
  patch that also adds the method `AttributedType::getAttr()` to get the
  type that an `AttributedType` instance is attributed with.
- Update `ports/llvm-16/portfile.cmake` to reflect this change.

Ninja3047 committed:
Drop llvm-15 + ubuntu 20.04 and more (#1029)

* Update vcpkg

* bump rellic

* bump anvill/remill

* Cleanup NuGet cache to fix space error on runner

* Use anvill patchable-ir-main

---------

Co-authored-by: Eric Kilmer <eric.d.kilmer@gmail.com>

Commit:
gap: bump up portfile
